storage/remote/queue_manager.go

```diff
@@ -218,9 +218,7 @@ func (t *QueueManager) Append(s *model.Sample) error {
 		return nil
 	}
 
-	t.shardsMtx.Lock()
 	enqueued := t.shards.enqueue(&snew)
-	t.shardsMtx.Unlock()
```

Comment:
IIUC, if you remove this lock its possible for this goroutine to read a stale value and do on an enqueue on a set of shards that has been stopped.

If you've found this to be a bottle neck you might want to test using a RWLock instead.  

I may be wrong, but at the very least I'd want to understand why you think removing this lock is safe?

Reply:
Sorry, I did not realize this problem and how to guarantee the samples. I will try to use a RWLock instead.